mcs/class/corlib/Mono/Runtime.cs

```diff
@@ -79,5 +80,29 @@ public static bool SetGCAllowSynchronousMajor (bool flag)
 			// No longer used
 			return true;
 		}
+
+#if !MOBILE 
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		static extern void DisableMicrosoftTelemetry (IntPtr appBundleID, IntPtr appSignature, IntPtr appVersion, IntPtr merpGUIPath);
+
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		static extern void EnableMicrosoftTelemetry_internal (IntPtr appBundleID, IntPtr appSignature, IntPtr appVersion, IntPtr merpGUIPath);
+
+		static void EnableMicrosoftTelemetry (string appBundleID_str, string appSignature_str, string appVersion_str, string merpGUIPath_str)
```

Comment:
Shouldn't this be public?

Reply:
These are invoked via dynamic invocation, none of the APIs here are surfaced on the public API, for compatibility with the surface area of .NET

Reply:
Then why the other APIs in the same class are public?

Reply:
This is not a general API and it's supported only on very specific configuration hence we keep it private.

The type is public on mobile platforms only where this API is not supported.